libraries/cyclestreets-view/src/main/java/net/cyclestreets/routing/Journey.kt

```diff
@@ -44,6 +46,42 @@ class Journey private constructor(wp: Waypoints? = null) {
     fun plan(): String { return start().plan() }
     fun speed(): Int { return start().speed() }
     fun totalDistance(): Int { return end().totalDistance() }
+    fun totalTime(): Int { return end().totalTime() }
+
+    fun remainingDistance(distanceUntilTurn: Int): Int {
+        val actSeg = activeSegment()
+
+        if (actSeg != null) {
+            return if (activeSegment > 0) (totalDistance() - actSeg.cumulativeDistance + distanceUntilTurn)
+                // segments[0] is a summary of the whole journey and its Cumulative distance is same as Total distance
+            else (totalDistance()) // Don't try and calculate until actual segment has been found
+        }
+        else { return Int.MAX_VALUE }
+    }
+
+    fun remainingTime(distanceUntilTurn: Int): Int {
+        val actSeg = activeSegment()
+        val prevSeg = previousSegment()
+        val prevSegTime : Int
+        val timeToEndOfSeg : Int
+
+        if (prevSeg != null) {prevSegTime = prevSeg.time}
+        else {prevSegTime = 0}
```

Comment:
this is not nice to read - it looks like a closure but it's just the body of an if statement.

Please refactor - look at code around.  We're not 100% consistent, but we don't have any code in that's this yucky 😂  I think this should be like:
```kotlin
prevSegTime = if (prevSeg != null) prevSeg.time else 0
```

In fact, if you do it this way, you can inline it with the declaration of `prevSegTime`.